File: src/1814.cpp
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int countNicePairs(vector<int>& nums) {
        long long pairs = 0;
        unordered_map<int, int> count;
        for (int i = 0; i < nums.size(); ++i) {
            nums[i] -= rev(nums[i]);
            pairs += count[nums[i]];
            count[nums[i]] += 1;
        }
        return pairs % (int) (1e9 + 7);
    }

    int rev(int n) {
        int r = 0;
        while (n > 0) {
            r *= 10;
            r += n % 10;
            n /= 10;
        }
        return r;
    }
};

File: src/1814.py
from collections import defaultdict

class Solution:
    def countNicePairs(self, nums: list[int]) -> int:
        def reverse(n):
            r = 0
            while n > 0:
                r *= 10
                r += n % 10
                n //= 10
            return r
        pairs = 0
        count = defaultdict(int)
        for i in range(len(nums)):
            nums[i] -= reverse(nums[i])
            pairs += count[nums[i]]
            count[nums[i]] += 1
        return pairs % (10**9 + 7)
